weasyl/favorite.py

```diff
@@ -254,7 +254,7 @@ def check(userid, submitid=None, charid=None, journalid=None):
         ], options="bool")
 
 
-def count(submitid):
+def count(submitid=None, journalid=None, charid=None):
     return d.execute(
-        "SELECT COUNT(*) FROM favorite WHERE targetid = %i AND type = 's'",
-        [submitid], options=['element'])
+        "SELECT COUNT(*) FROM favorite WHERE targetid = %i AND type = '%s'",
+        [submitid, 's' if submitid else 'f' if charid else 'j'], options=['element'])
```

Comment:
No worries. The 'obvious' errors are often the ones most easily overlooked. This is why we code review.